tiledb/sm/query/sparse_unordered_with_dups_reader.cc

```diff
@@ -1374,18 +1376,26 @@ Status SparseUnorderedWithDupsReader<BitmapType>::process_tiles(
         }
         buffers_full_ |= buffers_full;
 
+        // Clear tiles from memory and adjust result_tiles.
+        for (const auto& idx : *index_to_copy) {
+          const auto& name = names[idx];
+          if (condition_.field_names().count(name) == 0 &&
+              (!subarray_.is_set() || !is_dim)) {
+            clear_tiles(name, result_tiles, new_result_tiles_size);
```

Comment:
- IIUC we are using `new_result_tiles_size` as a minimum to clear the "rest" of the `result_tiles` before resizing...
- but the `clear_tiles` only happens conditionally, are there "other" tiles in `result_tiles` which are not cleared? Or are they simply known to be uninitialized based on this condition?

Reply:
There are two type of tiles we load in a result tile... Coordinate tiles (everything needed to select tiles, so "real" coordinate tiles if a subarray is set and query condition tiles) and attribute tiles. FYI we need to rename coordinate tiles to something else in this reader.

Coordinates tiles are loaded first and counted in the coords memory budget. They get unloaded at the end of every iteration if they are fully processed. Which will enable to possibly load more tiles on the next one. So these should not get unloaded here.

Attribute tiles are loaded during copy and counted in the copy memory budget. We load as many attributes in memory as we can in memory, copy, unload all, process next set of attributes. This clear tiles is for clearing attribute tiles only.

Reply:
From discussion, the piece I was missing here is that `result_tiles` is (practically speaking) a view of the underlying `result_tiles_` member variable -- so clearing here only removes items from the view, it does not destroy any tiles in memory.